docs/src/pages/guides/notifications.md

```diff
@@ -1,3 +1,189 @@
-# Notifications
+---
+id: notifications
+title: Notifications
+---
 
-under construction
+## Notification Types
+
+In Monika, currently we provided multiple notifications type. We are implementing email notifications through sendgrid, mailgun, smtp and webhook.
+
+## Configurations
+
+Notifications can be send through multiple processes and mail accounts, define through configurations as in the config.json.example.
+
+```json
+"notifications": [
+    {
+      "id": "unique-id",
+      "type": "mailgun",
+      "data": {
+        "recipients": ["mantap@jiwa.com"],
+        "apiKey": "YOUR_API_KEY",
+        "domain": "YOUR_DOMAIN"
+      }
+    },
+    {
+      "id": "unique-id",
+      "type": "sendgrid",
+      "data": {
+        "recipients": ["mantap@jiwa.com"],
+        "apiKey": "YOUR_API_KEY"
+      }
+    },
+    {
+      "id": "unique-id",
+      "type": "smtp",
+      "recipients": ["a@a.com"],
+      "data": {
+        "recipients": ["mantap@jiwa.com"],
+        "hostname": "https://www.dennypradipta.com",
+        "port": 8080,
+        "username": "dennypradipta",
+        "password": "bismillah"
+      }
+    },
+    {
+      "id": "unique-id-webhook",
+      "type": "webhook",
+      "data": {
+        "recipients": ["mantap@jiwa.com"],
+        "method": "POST",
+        "url": "https://examplewebhookurl.com/webhook"
+      }
+    }
+  ],
+```
+
+## Mailgun
+
+Mailgun is the notification send through a mailgun email service. For this you need to have a mailgun account first, using your mailgun's API_KEY and DOMAIN you can send notifications to the recipient's mail.
+
+```json
+{
+  "id": "unique-id",
+  "type": "mailgun",
+  "data": {
+    "recipients": ["mantap@jiwa.com"],
+    "apiKey": "YOUR_API_KEY",
+    "domain": "YOUR_DOMAIN"
+  }
+}
+```
+
+Monika is filtering the notification type for mailgun and breaking down the data config. Setting a mailgun service and then send through the data using mailgun-js lib.
+
+```js
+const mg = mailgun({ apiKey: API_KEY, domain: DOMAIN })
+const data = {
+  from: `${sender.name} <${sender.email}>`,
+  to: recipients,
+  subject: subject,
+  text: body,
+}
+return mg.messages().send(data)
+```
+
+## Sendgrid
+
+As in the mailgun, sendgrid is also an email delivery service. You need to have a sendgrid account before you can send an email through this method. For sending a notification through sendgrid all you need to set is the API_KEY and recipient email addresses.
+
+```json
+{
+  "id": "unique-id",
+  "type": "sendgrid",
+  "data": {
+    "recipients": ["mantap@jiwa.com"],
+    "apiKey": "YOUR_API_KEY"
+  }
+}
+```
+
+Monika is using @sendgrid/mail library for sending the sendgrid type notification. Monika will breakdown the configuration set the sengrid type API_KEY settings and send the message using the library.
+
+```js
+sgMail.setApiKey(API_KEY)
+const msg = {
+  to: recipients,
+  from: sender.email,
+  subject,
+  text: body,
+}
+
+sgMail.send(msg)
+```
```

Comment:
I think this part doesn't need to be in the guides. User doesn't need to know our implementation detail, just need to know how to use

Reply:
modified